functions/Find-DbaDatabase.ps1

```diff
@@ -0,0 +1,200 @@
+You should have received a copy of the GNU General Public License along with this program. If not, see http://www.gnu.org/licenses/.
+
+.LINK
+ https://dbatools.io/Find-DbaDatabase
+
+.EXAMPLE
+Find-DbaDatabase -SqlInstance "DEV01", "DEV02", "UAT01", "UAT02", "PROD01", "PROD02" -Type Database -Name TestDB -Detailed 
+Returns all database from the SqlInstances that have a database named TestDB with a detailed output.
+
+.EXAMPLE
+Find-DbaDatabase -SqlInstance "DEV01", "DEV02", "UAT01", "UAT02", "PROD01", "PROD02" -Type Service_Broker_GUID -Name 25b64fef-faeb-495a-9898-f25a782835f5 -Detailed 
+Returns all database from the SqlInstances that have the same Service Broker GUID with a deeatiled output
+
+#>
+	[CmdletBinding()]
+	param (
+		[Parameter(Mandatory = $true, ValueFromPipeline = $true)]
+		[Alias('cn','host','computer','server')]
+		[string[]]$SqlInstance,
+		[System.Management.Automation.PSCredential]$SqlCredential,
+        [parameter(Mandatory = $true)]
+        [ValidateSet('Database','Service_Broker_GUID')]
+        [string]$Type,
+        [parameter(Mandatory = $true)]
+        [string]$Name,
+        [switch]$Like,
+        [switch]$Detailed
+	)
+    process
+    {
+        foreach ($Instance in $SqlInstance)
+        {
+            try
+	        {
+	            Write-Verbose "Connecting to $Instance"
+                $srv = Connect-SqlServer -SqlServer $Instance -SqlCredential $sqlcredential
+	        }
+	        catch
+	        {
+	            Write-Warning "Failed to connect to: $srv"
+                break
+	        }
+
+            if($srv -contains "\")
+            {
+                $computername = ($srv.Name).Split("\")[0]
+                $inst =  ($srv.Name).Split("\")[1]
+            }
+            else
+            {
+                $computername = $srv.Name
+                $inst =  ""
+            }
+
+        if ($Type -eq 'Database')
+        {
+            if ($Like)
+            {
+                $match = $srv.Databases | Where-Object {$_.Name -like "*$name*"}
+            }
+            else
+            {
+                $match = $srv.Databases | Where-Object {$_.Name -eq $name}
+            }
+            foreach ($db in $match)
+            {
+                if ($Detailed)
+                {
+                    if ($db.ExtendedProperties.Count -ne 0)
+                    {
+                        $outep = @()
+                        foreach ($xp in $db.ExtendedProperties) 
+                        {
+                            $extdetails = [PSCustomObject]@{
+		                       Name = $db.ExtendedProperties[$xp.Name].Name
+		                       Value = $db.ExtendedProperties[$xp.Name].Value
+                                }
+                            $outep += $extdetails
+                        }
+                    }
+                    else
+                    {
+                        $outep = ""
+                    }
+                    $out = [PSCustomObject]@{
```

Comment:
No need to assign the variable since it needs to manipulation after (like with Select-DefaultItem

Just send [PSCustomObject]@{ right now to the pipeline